File: Back/src/ProEventos.Domain/Event.cs
using System;
using System.Collections.Generic;

namespace ProEventos.Domain
{
    public class Event
    {
        public int Id { get; set; }
        public string Location { get; set; }
        public DateTime? EventDate { get; set; }
        public string Theme { get; set; }
        public int NumberOfPeople { get; set; }
        public string ImageURL { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public IEnumerable<Batch> Batches { get; set; }
        public IEnumerable<SocialNetwork> SocialNetworks { get; set; }
        public IEnumerable<SpeecherEvent> SpeecherEvents { get; set; }

    }
}

File: Front/ProEventos-App/src/app/models/Event.ts
import { Batch } from './Batch';
import { SocialNetwork } from './SocialNetwork';
import { Speecher } from './Speecher';

export interface Event {
  id: number;
  location: string;
  eventDate?: Date;
  theme: string;
  numberOfPeople: number;
  imageURL: string;
  phoneNumber: string;
  email: string;
  batches: Batch[];
  socialNetworks: SocialNetwork[];
  speecherEvents: Speecher[];

  // id?: number;
  // location?: string;
  // eventDate?: Date;
  // theme?: string;
  // numberOfPeople?: number;
  // imageURL?: string;
  // phoneNumber?: string;
  // email?: string;
  // batches?: Batch[];
  // socialNetworks?: SocialNetwork[];
  // speecherEvents?: Speecher[];


  // constructor(id: number, location: string, eventDate: Date, theme: string, numberOfPeople: number, imageURL: string,
  //             phoneNumber: string, email: string, batches: Batch[], socialNetworks: SocialNetwork[], speecherEvents: Speecher[]) {
  //   this.id = id;
  //   this.location = location;
  //   this.eventDate = eventDate;
  //   this.theme = theme;
  //   this.numberOfPeople = numberOfPeople;
  //   this.imageURL = imageURL;
  //   this.phoneNumber = phoneNumber;
  //   this.email = email;
  //   this.batches = batches;
  //   this.socialNetworks = socialNetworks;
  //   this.speecherEvents = speecherEvents;
  // }
}
